File: app/src/main/java/com/parabola/newtone/di/app/AppComponentInjects.java
package com.parabola.newtone.di.app;

import com.parabola.newtone.mvp.presenter.AlbumPresenter;
import com.parabola.newtone.mvp.presenter.ArtistPresenter;
import com.parabola.newtone.mvp.presenter.ArtistTracksPresenter;
import com.parabola.newtone.mvp.presenter.ChoosePlaylistPresenter;
import com.parabola.newtone.mvp.presenter.CreatePlaylistPresenter;
import com.parabola.newtone.mvp.presenter.DeletePlaylistPresenter;
import com.parabola.newtone.mvp.presenter.FavouritesPlaylistPresenter;
import com.parabola.newtone.mvp.presenter.FolderPresenter;
import com.parabola.newtone.mvp.presenter.FoldersListPresenter;
import com.parabola.newtone.mvp.presenter.MainPresenter;
import com.parabola.newtone.mvp.presenter.PlayerPresenter;
import com.parabola.newtone.mvp.presenter.PlaylistPresenter;
import com.parabola.newtone.mvp.presenter.QueuePresenter;
import com.parabola.newtone.mvp.presenter.RecentlyAddedPlaylistPresenter;
import com.parabola.newtone.mvp.presenter.RenamePlaylistPresenter;
import com.parabola.newtone.mvp.presenter.SearchPresenter;
import com.parabola.newtone.mvp.presenter.SettingPresenter;
import com.parabola.newtone.mvp.presenter.SleepTimerPresenter;
import com.parabola.newtone.mvp.presenter.StartPresenter;
import com.parabola.newtone.mvp.presenter.TabAlbumPresenter;
import com.parabola.newtone.mvp.presenter.TabArtistPresenter;
import com.parabola.newtone.mvp.presenter.TabPlaylistPresenter;
import com.parabola.newtone.mvp.presenter.TabTrackPresenter;
import com.parabola.newtone.mvp.presenter.TimeToSleepInfoPresenter;
import com.parabola.newtone.mvp.presenter.TrackAdditionalInfoPresenter;
import com.parabola.newtone.mvp.presenter.fx.FxAudioSettingsPresenter;
import com.parabola.newtone.mvp.presenter.fx.TabEqualizerPresenter;
import com.parabola.newtone.ui.activity.MainActivity;
import com.parabola.newtone.ui.dialog.SortingDialog;
import com.parabola.newtone.ui.fragment.settings.AlbumItemDisplaySettingFragment;
import com.parabola.newtone.ui.fragment.settings.ArtistItemDisplaySettingFragment;
import com.parabola.newtone.ui.fragment.settings.TrackItemDisplaySettingFragment;

public interface AppComponentInjects {

    void inject(MainActivity activity);

    void inject(MainPresenter presenter);

    //    FIRST SCREENS
    void inject(StartPresenter presenter);
    void inject(TabArtistPresenter presenter);
    void inject(TabAlbumPresenter presenter);
    void inject(TabTrackPresenter presenter);
    void inject(TabPlaylistPresenter presenter);


    void inject(ArtistPresenter presenter);
    void inject(AlbumPresenter presenter);
    void inject(ArtistTracksPresenter presenter);
    void inject(PlayerPresenter presenter);
    void inject(FolderPresenter presenter);
    void inject(SearchPresenter presenter);


    //  SETTINGS
    void inject(SettingPresenter presenter);
    void inject(TrackItemDisplaySettingFragment fragment);
    void inject(AlbumItemDisplaySettingFragment fragment);
    void inject(ArtistItemDisplaySettingFragment fragment);


    //  PLAYLISTS
    void inject(PlaylistPresenter presenter);
    void inject(FavouritesPlaylistPresenter presenter);
    void inject(RecentlyAddedPlaylistPresenter presenter);
    void inject(QueuePresenter presenter);
    void inject(FoldersListPresenter presenter);


    //    DIALOGS
    void inject(CreatePlaylistPresenter presenter);
    void inject(RenamePlaylistPresenter presenter);
    void inject(DeletePlaylistPresenter presenter);
    void inject(SleepTimerPresenter presenter);
    void inject(TimeToSleepInfoPresenter presenter);
    void inject(ChoosePlaylistPresenter presenter);
    void inject(SortingDialog dialog);
    void inject(TrackAdditionalInfoPresenter presenter);


    //    FXs
    void inject(TabEqualizerPresenter presenter);
    void inject(FxAudioSettingsPresenter presenter);
}

File: app/src/main/java/com/parabola/newtone/di/app/AppComponentInjects.kt
package com.parabola.newtone.di.app

import com.parabola.newtone.presentation.audioeffects.settings.FxAudioSettingsPresenter
import com.parabola.newtone.presentation.audioeffects.equalizer.FxEqualizerPresenter
import com.parabola.newtone.presentation.album.AlbumPresenter
import com.parabola.newtone.presentation.artist.ArtistPresenter
import com.parabola.newtone.presentation.artisttracks.ArtistTracksPresenter
import com.parabola.newtone.presentation.playlist.chooseplaylist.ChoosePlaylistPresenter
import com.parabola.newtone.presentation.playlist.createplaylist.CreatePlaylistPresenter
import com.parabola.newtone.presentation.folder.FolderPresenter
import com.parabola.newtone.presentation.main.start.StartPresenter
import com.parabola.newtone.presentation.main.albums.TabAlbumPresenter
import com.parabola.newtone.presentation.main.artists.TabArtistPresenter
import com.parabola.newtone.presentation.main.playlists.TabPlaylistPresenter
import com.parabola.newtone.presentation.main.tracks.TabTrackPresenter
import com.parabola.newtone.presentation.player.PlayerPresenter
import com.parabola.newtone.presentation.playlist.favourites.FavouritesPlaylistPresenter
import com.parabola.newtone.presentation.playlist.folderslist.FoldersListPresenter
import com.parabola.newtone.presentation.playlist.playlist.PlaylistPresenter
import com.parabola.newtone.presentation.playlist.queue.QueuePresenter
import com.parabola.newtone.presentation.playlist.recentlyadded.RecentlyAddedPlaylistPresenter
import com.parabola.newtone.presentation.search.SearchPresenter
import com.parabola.newtone.presentation.mainactivity.MainActivity
import com.parabola.newtone.presentation.mainactivity.MainPresenter
import com.parabola.newtone.presentation.DeleteTrackDialog
import com.parabola.newtone.presentation.player.sleeptimer.SleepTimerPresenter
import com.parabola.newtone.presentation.player.timetosleepinfo.TimeToSleepInfoPresenter
import com.parabola.newtone.presentation.playlist.renameplaylist.RenamePlaylistPresenter
import com.parabola.newtone.presentation.SortingDialog
import com.parabola.newtone.presentation.audioeffects.EqPresetsSelectorDialog
import com.parabola.newtone.presentation.settings.AlbumItemDisplaySettingFragment
import com.parabola.newtone.presentation.settings.ArtistItemDisplaySettingFragment
import com.parabola.newtone.presentation.settings.ExcludedFoldersFragment
import com.parabola.newtone.presentation.settings.SettingPresenter
import com.parabola.newtone.presentation.settings.TrackItemDisplaySettingFragment
import com.parabola.newtone.presentation.settings.colorthemeselector.ColorThemeSelectorPresenter
import com.parabola.newtone.presentation.trackadditionalinfo.TrackAdditionalInfoPresenter

interface AppComponentInjects {
    fun inject(activity: MainActivity)
    fun inject(presenter: MainPresenter)

    //    FIRST SCREENS
    fun inject(presenter: StartPresenter)
    fun inject(presenter: TabArtistPresenter)
    fun inject(presenter: TabAlbumPresenter)
    fun inject(presenter: TabTrackPresenter)
    fun inject(presenter: TabPlaylistPresenter)
    fun inject(presenter: ArtistPresenter)
    fun inject(presenter: AlbumPresenter)
    fun inject(presenter: ArtistTracksPresenter)
    fun inject(presenter: PlayerPresenter)
    fun inject(presenter: FolderPresenter)
    fun inject(presenter: SearchPresenter)

    //  SETTINGS
    fun inject(presenter: SettingPresenter)
    fun inject(presenter: ColorThemeSelectorPresenter)
    fun inject(fragment: ExcludedFoldersFragment)
    fun inject(fragment: TrackItemDisplaySettingFragment)
    fun inject(fragment: AlbumItemDisplaySettingFragment)
    fun inject(fragment: ArtistItemDisplaySettingFragment)

    //  PLAYLISTS
    fun inject(presenter: PlaylistPresenter)
    fun inject(presenter: FavouritesPlaylistPresenter)
    fun inject(presenter: RecentlyAddedPlaylistPresenter)
    fun inject(presenter: QueuePresenter)
    fun inject(presenter: FoldersListPresenter)

    //    DIALOGS
    fun inject(presenter: CreatePlaylistPresenter)
    fun inject(presenter: RenamePlaylistPresenter)
    fun inject(presenter: SleepTimerPresenter)
    fun inject(presenter: TimeToSleepInfoPresenter)
    fun inject(presenter: ChoosePlaylistPresenter)
    fun inject(dialog: SortingDialog)
    fun inject(dialog: EqPresetsSelectorDialog)
    fun inject(presenter: TrackAdditionalInfoPresenter)
    fun inject(dialog: DeleteTrackDialog)

    //    FXs
    fun inject(presenter: FxEqualizerPresenter)
    fun inject(presenter: FxAudioSettingsPresenter)
}
